source/hp/dof_handler.cc

```diff
@@ -2321,15 +2321,10 @@ namespace hp
 
                           if (i == (*finite_elements)[fe_index_1].dofs_per_line)
                             {
-                              // the dofs of these two finite elements are
-                              // identical. as a safety check, ensure that none
-                              // of the two FEs is trying to dominate the other,
-                              // which wouldn't make any sense in this case
-                              Assert ((*finite_elements)[fe_index_1].compare_for_face_domination
-                                      ((*finite_elements)[fe_index_2])
-                                      ==
-                                      FiniteElementDomination::either_element_can_dominate,
-                                      ExcInternalError());
+                              // the line dofs of these two finite elements are identical.
+                              // Note that there could be situations when one element still dominates another, e.g.:
+                              // FE_Q(2) x FE_Nothing(dominate) vs
+                              // FE_Q(2) x FE_Q(1)
```

Comment:
i think they do. Because there is a difference between line dofs and face dofs (line+vertex).
My understanding is that FE_Q has no line dofs and therefore the two have the same number (1 in 2d) of line dofs.